### PR TITLE
Changing "proxy-endpoints" flow

### LINKED ISSUE #567: Paypal Redirect issue once you cancel or place and order
**Describe the bug**
When a user places an order through paypal or cancels, the user should be redirected to their storefront.

The user on the current demo gets a 'the site cant be reached' page
On my staging site I get json code, which maybe related to this issue.  See screenshots below

**To Reproduce**
1. Get to checkout and checkout with paypal
2. Cancel the order on paypal or fill in credit card details and place the order. 
3. You should get the error message with the rest api in the address bar.

**Expected behavior**
User should be redirected back to the store front for both cancel and successful orders

**Screenshots**
Demo site

screen_shot_2019-08-28_at_21 25 10

My staging site
Screen Shot 2019-08-29 at 11 00 01

Just receive json response instead of a redirect. 

**Additional context**
I've tried with both guest and logged in users.

### PR DESCRIPTION
**Pull request by Borales**
### Please check if the PR fulfills these requirements ###

- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Bugfix

**What is the current behavior? (You can also link to an open issue here)**
After enabling full-page "cache" (via service-worker) - the proxied endpoints were left behind the cache by mistake.

**What is the new behavior (if this is a feature change)?**
All "proxied" endpoints that we handle on Falcon-Client must be processed without any cache calls in the browser. This PR defines a `blacklist` of routes that will be excluded from the generated `sw.js` file. Due to the nature of building `sw.js` file (mostly for production use) - this list of routes must be available inside of Falcon-Client application (to be able to build the required files using CI/CD), thus we cannot fetch this list dynamically anymore - this PR changes this behavior. 

> It is recommended to perform a manual synchronization of these endpoints by checking `/config` endpoint on Falcon-Server app (for example, `http://localhost:4000/config` when running locally) and copying `endpoints` value to the Falcon-Client config (`proxyEndpoints` root key).

**Does this PR introduce a breaking change? (What changes might users need to make in their application due to this PR?)**
Yes, it does. 

`client/falcon-client.build.config.js`:

```diff
  serviceWorker: {
-    precache: process.env.NODE_ENV === 'production'
+    precache: process.env.NODE_ENV === 'production',
+    blacklistRoutes: config.proxyEndpoints || []
  },
```

`client/bootstrap.js`:

```diff
-    onRouterCreated: async router => configureProxy(router, serverUrl, serverConfig.endpoints, redirects)
+    onRouterCreated: async router => configureProxy(router, serverUrl, config.proxyEndpoints, redirects)
```

**Other information:**
Fixes #567

To verify that it works - you should see the following console message about "blacklist pattern" when using such proxied endpoints:

Screenshot 2019-09-03 at 15 48 29

### CODE REVIEW

File: packages/falcon-client/src/serviceWorker/sw.js [hunk 1]
```diff
@@ -80,22 +82,27 @@ async function getFromCacheOrNetwork(request) {
...
+
+      const cachedUrlKey = getCacheKeyForURL('app-shell');
+      if (!cachedUrlKey) {
+        return fetch(url.href);
+      }
+
+      return getFromCacheOrNetwork(cachedUrlKey);
+    },
+    {
+      blacklist: BLACKLIST_ROUTES.map(route => new RegExp(route))
```
Comment (tutok) [inline packages/falcon-client/src/serviceWorker/sw.js]: You have already entire config stored in `CONFIG` variable, so `BLACKLIST_ROUTES` is not necessary, please remove it

```suggestion
      blacklist: CONFIG.blacklistRoutes.map(route => new RegExp(route))
```

File: packages/falcon-client/build-utils/serviceWorker/index.js [hunk 2]
```diff
@@ -10,6 +10,16 @@ const alias = require('rollup-plugin-alias');
 const paths = require('../paths');
 const { getManifestEntries } = require('./workbox');
 
+const getPatternList = (NODE_ENV, buildConfig, manifestEntries) => [
+  { test: 'process.env.NODE_ENV', replace: JSON.stringify(NODE_ENV) },
+  { test: 'const CONFIG = {};', replace: `const CONFIG = ${JSON.stringify(buildConfig, null, 2)};` },
+  { test: 'const ENTRIES = [];', replace: `const ENTRIES = ${JSON.stringify(manifestEntries, null, 2)};` },
+  {
+    test: 'const BLACKLIST_ROUTES = [];',
```
Comment (tutok) [inline packages/falcon-client/build-utils/serviceWorker/index.js]: `BLACKLIST_ROUTES` is not necessary, please remove it.

File: packages/falcon-client/build-utils/serviceWorker/index.js [hunk 3]
```diff
@@ -10,6 +10,16 @@ const alias = require('rollup-plugin-alias');
 const paths = require('../paths');
 const { getManifestEntries } = require('./workbox');
 
+const getPatternList = (NODE_ENV, buildConfig, manifestEntries) => [
```
Comment (tutok) [inline packages/falcon-client/build-utils/serviceWorker/index.js]:
```suggestion
const getRePatternList = (NODE_ENV, buildConfig, manifestEntries) => [
```

File: packages/falcon-client/src/serviceWorker/sw.js (hunk 1 above)
Comment (Borales) [reply packages/falcon-client/src/serviceWorker/sw.js]: done

File: packages/falcon-client/build-utils/serviceWorker/index.js (hunk 2 above)
Comment (Borales) [reply packages/falcon-client/build-utils/serviceWorker/index.js]: done

File: packages/falcon-client/build-utils/serviceWorker/index.js (hunk 3 above)
Comment (Borales) [reply packages/falcon-client/build-utils/serviceWorker/index.js]: done

**Pull request merged**